core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/api/TrackNetworkInfra.kt

```diff
@@ -27,12 +27,14 @@ typealias TrackNodePortId = StaticIdx<TrackNodePort>
 sealed interface TrackNodeConfig
 typealias TrackNodeConfigId = StaticIdx<TrackNodeConfig>
 
-
+@Suppress("INAPPLICABLE_JVM_NAME")
 interface TrackNetworkInfra {
-    fun getNextTrackSection(current: DirTrackSectionId, config: TrackNodeConfigId): OptDirTrackSectionId
+    fun getNextTrackSection(current: EndpointTrackSectionId, config: TrackNodeConfigId): OptDirTrackSectionId
+    @JvmName("getNextTrackSections")
+    fun getNextTrackSections(trackSection: DirTrackSectionId): DirStaticIdxList<TrackSection>
 
-    fun getNextTrackNode(trackSection: DirTrackSectionId): OptStaticIdx<TrackNode>
-    fun getNextTrackNodePort(trackSection: DirTrackSectionId): OptStaticIdx<TrackNodePort>
+    fun getNextTrackNode(trackEndpoint: EndpointTrackSectionId): OptStaticIdx<TrackNode>
+    fun getNextTrackNodePort(trackEndpoint: EndpointTrackSectionId): OptStaticIdx<TrackNodePort>
```

Comment:
The interface isn't really consistent there. I'd prefer handling track endpoints over directions in this context, when mixing the two versions there's a risk for mistakes (especially since ids aren't typed in java).

Reply:
Yes I kinda hesitated putting tack endpoint in `getNextTrackSections` as well, given the fact that's it's a method called in java, as such i'd need to create the corresponding EndpointTrackSectionId somehow. But i do agree with you. I'll just add a static method to get the `EnpointTrackSectionId` from a `DirTrackSectionId` and do the corresponding changes.